### PR TITLE
Use awesome `Steps` component in tutorial

### PR DESCRIPTION
**Pull request by liruifengv**
#### Description (required)

Use awesome `Steps` component in our blog tutorial

#### Related issues & labels (optional)

- Closes #
- Suggested label:

### CODE REVIEW

**Review by casungo — COMMENTED** [review]
I left some comments with screenshots to mark some places where the steps may not be used correctly

File: src/content/docs/en/tutorial/5-astro-api/2.mdx
[the remark this answers is not in the document]
Comment (liruifengv) [reply src/content/docs/en/tutorial/5-astro-api/2.mdx:1]: The <Steps> component can't used for this. Only can wrap  a single ordered list (<ol>

**Comment by liruifengv** [comment]
Thanks for your review @casungo Do you have any good ideas for a list with only one item?

**Review by morinokami — COMMENTED** [review]
I think we can add `Steps` to the following sections as well:

- https://github.com/liruifengv/docs/blob/steps-tutorial/src/content/docs/en/tutorial/1-setup/index.mdx#L28-L46
- https://github.com/liruifengv/docs/blob/steps-tutorial/src/content/docs/en/tutorial/5-astro-api/1.mdx#L104-L174
- https://github.com/liruifengv/docs/blob/steps-tutorial/src/content/docs/en/tutorial/5-astro-api/3.mdx#L44-L76

---

IMHO, while it may not be strictly correct to use the `Steps` component for a single step, since this doesn't technically constitute multiple 'steps', I believe it is still acceptable. Maintaining consistency in using the `Steps` component, even for a single step, should help users understand that this is an instruction they should follow.

File: src/content/docs/en/tutorial/5-astro-api/2.mdx
[the remark this answers is not in the document]
Comment (casungo) [reply src/content/docs/en/tutorial/5-astro-api/2.mdx:1]: oh ok thanks for the insight!

**Review by VoxelMC — COMMENTED** [review]
Pretty nifty work! The Steps component really brings the visuals to the next level! Thank you for taking on this job (emphasis on **job**).

I have left a few of my opinions, I hope they help! :)

**Comment by liruifengv** [comment]
> I think we can add `Steps` to the following sections as well:
> 
> * https://github.com/liruifengv/docs/blob/steps-tutorial/src/content/docs/en/tutorial/1-setup/index.mdx#L28-L46
> * https://github.com/liruifengv/docs/blob/steps-tutorial/src/content/docs/en/tutorial/5-astro-api/1.mdx#L104-L174
> * https://github.com/liruifengv/docs/blob/steps-tutorial/src/content/docs/en/tutorial/5-astro-api/3.mdx#L44-L76
> 

Done

File: src/content/docs/en/tutorial/2-pages/4.mdx [hunk 1]
```diff
@@ -128,6 +131,7 @@ The Astro `<style>` tag can also reference any variables from your frontmatter s
     ```
 
 3. Check your About page in your browser preview. You should see that the skills are now navy blue, as set by the `skillColor` variable passed to the `define:vars` directive.
+</Steps>
 
 <Box icon="puzzle-piece">
```
[the remark this answers is not in the document]
Comment (liruifengv) [reply src/content/docs/en/tutorial/2-pages/4.mdx:137]: Yeah!

File: src/content/docs/en/tutorial/4-layouts/1.mdx [hunk 2]
```diff
@@ -53,9 +54,11 @@ You still have some Astro components repeatedly rendered on every page. It's tim
       </body>
     </html>
     ```
+</Steps>
 
 ## Use your layout on a page
 
+<Steps>
 3. Replace the code at `src/pages/index.astro` with the following:
```
[the remark this answers is not in the document]
Comment (liruifengv) [reply src/content/docs/en/tutorial/4-layouts/1.mdx:62]: Ok. I update it. see: https://github.com/withastro/docs/pull/7496/commits/04b579cda140cb040f210291540297c18208b2cb

File: src/content/docs/en/tutorial/2-pages/3.mdx [hunk 3]
```diff
@@ -13,8 +13,7 @@ import Box from '~/components/tutorial/Box.astro';
 import MultipleChoice from '~/components/tutorial/MultipleChoice.astro';
 import Option from '~/components/tutorial/Option.astro';
 import PreCheck from '~/components/tutorial/PreCheck.astro';
-
-
+import { Steps } from '@astrojs/starlight/components';
 
 Now that you have a multi-page website with HTML content, it's time to add some dynamic HTML!
```
[the remark this answers is not in the document]
Comment (liruifengv) [reply src/content/docs/en/tutorial/2-pages/3.mdx:18]: Which line? I don't see.

File: src/content/docs/en/tutorial/1-setup/2.mdx [hunk 4]
```diff
@@ -63,11 +63,13 @@ The preferred way to create a new Astro site is through our `create astro` setup
...
+</Steps>
 
 When the install wizard is complete, you no longer need this terminal. You can now open VS Code to continue.
 
 ## Open your project in VS Code
 
+<Steps>
 1. Open VS Code. You will be prompted to open a folder. Choose the folder that you created during the setup wizard.
 
 2. If this is your first time opening an Astro project, you should see a notification asking if you would like to install recommended extensions. Click to see the recommended extensions, and choose the [Astro language support extension](https://marketplace.visualstudio.com/items?itemName=astro-build.astro-vscode). This will provide syntax highlighting and autocompletions for your Astro code.
```
[the remark this answers is not in the document]
Comment (liruifengv) [reply src/content/docs/en/tutorial/1-setup/2.mdx]: It's great!

File: src/content/docs/en/tutorial/6-islands/1.mdx
[the remark this answers is not in the document]
Comment (liruifengv) [reply src/content/docs/en/tutorial/6-islands/1.mdx:1]: I don't think the `questions` should be the steps

File: src/content/docs/en/tutorial/1-setup/2.mdx [hunk 5]
```diff
@@ -92,6 +95,7 @@ In order to preview your project files _as a website_ while you work, you will n
 
 ### Start the dev server
 
+<Steps>
```
[the remark this answers is not in the document]
Comment (liruifengv) [reply src/content/docs/en/tutorial/1-setup/2.mdx:98]: I changed the listitem number in this page. see: https://github.com/withastro/docs/pull/7496/commits/1aae53c70381863db75e02a6bf0dc2052d72f026

**Comment by liruifengv** [comment]
To summarise the problem:
- How to deal with only one listitem in page.
- Only one listitem in Box
image
- The listitem in the page are all in order. See: https://github.com/withastro/docs/pull/7496/commits/1aae53c70381863db75e02a6bf0dc2052d72f026

File: src/content/docs/en/tutorial/6-islands/2.mdx [hunk 6]
```diff
@@ -68,11 +68,13 @@ Let's build a clickable icon to let your users toggle between light or dark mode
     ```
 
 3. Visit your browser preview at `http://localhost:4321` to see the icon now on all your pages. You can try clicking it, but you have not written a script to make it interactive yet.
+</Steps>
 
 ## Add CSS styling for a dark theme
 
 Choose some alternate colors to use in dark mode.
 
+<Steps>
```
[the remark this answers is not in the document]
Comment (liruifengv) [reply src/content/docs/en/tutorial/6-islands/2.mdx:77]: Maybe this page could also increment the list item numbers sequentially?


like: https://github.com/withastro/docs/commit/1aae53c70381863db75e02a6bf0dc2052d72f026

**Comment by sarah11918** [comment]
Resolved all the outstanding issues on Talking and Doc'ing!

In summary, yes, one item lists are good! (They call out when you have to do something.)

And yes, even one item lists in a box are good! (same reasoning).

**Review by at-the-vr — APPROVED** [review]
I was in T&D 😄 + love the changes LGTM 👍

**Comment by Fryuni** [comment]
I listened to it happening! It sounded nice, hahaha

**Review by mingjunlu — APPROVED** [review]
LGTM! Nice work everyone 👏

**Review by sarah11918 — APPROVED** [review]
We ALL approve! Talking and Doc'ing says LGTM! Thank you for this amazing PR, @liruifengv !

**Comment by liruifengv** [comment]
Thank you everyone!💗

**Review by casungo — APPROVED** [review]
Couldn't be present at this Talking and Doc'ing sadly.
LGTM :)

**Comment by sarah11918** [comment]
!coauthor

File: src/content/docs/en/tutorial/2-pages/3.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (VoxelMC) [reply src/content/docs/en/tutorial/2-pages/3.mdx:18]: I no longer remember, seems like I misclicked here haha

**Pull request merged**